### PR TITLE
New search experience via Swiftype

### LINKED ISSUE #4642: [Epic] Better docs search: GA
## overview
current docs search is pretty broken and does not meet users needs. some previous convo around this happened in #1558 and #4213

### examples of user's frustration
- searching for `transformation` gives some results, none of which link you to the resource options which was what the user was looking for
- searching for `tokens` links you to 5 different pages on tokens none with any context around what questions they answer
    docs-search-tokens
- trying to search for specific features in the console leads to frustration usually with no relevant results

## features we need in search
- [ ] accurate results
- [ ] typos should not result in limited results
- [ ] search results need context to the information they provide, e.g., is it part of the aws docs?
- [ ] auto-complete, and ability to see results as you type

## tools to research
- [ ] [algolia docs search](https://docsearch.algolia.com/)
- [ ] [swift type](https://swiftype.com/)
- [ ] [elasticsearch](https://www.elastic.co/elasticsearch/)

### LINKED ISSUE #5250: move docs search
### Problem description

with the improvements to docs search, it be cool to also consider moving search to a more prominent location so its super clear to everyone that we have docs search; im including a few explorations, super open to feedback or other ideas

#### top-level
across the top above all the content, we could even go full width; would be nice to keep this sticky at the top if possible
top-level

#### top of left-nav instead of right-aside
this is another location which wouldn't require us to make it sticky, but wed still have the width issue; also a nit the left nav should really be in a nav element instead of an aside element
docs-left-nav

### PR DESCRIPTION
**Pull request by davidwrede**
This PR changes our site search experience from using lunr.js to the managed service Swiftype Site Search.

This should provide a richer and better site search experience than we have now with more relevant and useful information returned in result sets to help you get to the doc page you're looking for faster.

To test out the experience, click on the staging site URL for the most recent commit. The search bar is currently on the top-right of any /docs page next to the right nav. You can test out both the autocomplete experience through the search bar and clicking on a result, or to get more detailed information in the result set, hit enter after typing in your search query to pick a selection from the modal dialog that pops up.

**Note**: Until this change is merged in, whenever you click on a result from the result set, it will take you out to the production site (which still uses the current site search experience).

Fixes #4642
Fixes #5250

Items left to be done:

- [x] Settle on scrolling experience (See https://github.com/pulumi/docs/pull/5223#issuecomment-778375604)
- [x] Setup correct billing info in Swiftype admin console
- [x] Tweak/optimize result sets for the top 10-15 highly-viewed and/or important topics

### CODE REVIEW

**Comment by susanev** [comment]
if you happen to scroll while searching the results scroll right off the page, that seems not correct

search-results

File: scripts/build-site.sh [hunk 1]
```diff
@@ -35,7 +35,4 @@ yarn run tsc --outFile ${JS_BUNDLE}
 printf "Compiling Sass and running PostCSS...\n\n"
 yarn run --silent node-sass assets/sass/styles.scss | yarn run postcss --config assets/config --output ${CSS_BUNDLE}
 
-printf "Building the search index...\n\n"
-make build_search_index
```
Comment (justinvp) [inline scripts/build-site.sh:39]: We can also remove the `build_search_index` target from the `Makefile`:

https://github.com/pulumi/docs/blob/26a033fb30a38ebc19eeac0b395ad381c597b713/Makefile#L48-L52

File: layouts/partials/docs/search.html [hunk 2]
```diff
@@ -1,9 +1,15 @@
-<form class="relative" action="{{ relref . "/docs/search" }}" method="GET">
-    <i class="fas fa-search absolute inset-y-0 flex items-center pl-3 text-gray-500"></i>
+<script type="text/javascript">
```
Comment (justinvp) [inline layouts/partials/docs/search.html:1]: Nit: No need to specify `type="text/javascript"`:

```suggestion
<script>
```

File: layouts/docs/search.html [hunk 3]
```diff
@@ -1,30 +0,0 @@
...
-        </div>
-    </div>
-
-    <!--
-        Minify and fingerprint script. Generated file will contain content hash in
-        filename, so no need to add query param to cache bust. Also note that because this
-        script references functions defined in the main bundle, we defer loading it until
-        that script's been evaluated.
-    -->
-    {{ $js := resources.Get "js/search.js" | resources.Minify | resources.Fingerprint }}
```
Comment (justinvp) [inline layouts/docs/search.html:28]: `assets/js/search.js` and `static/js/search-worker.js` can be deleted.

If I'm remembering correctly, `assets/sass/_search.scss` can also be deleted, along with the import for it in `assets/sass/styles.scss`:

https://github.com/pulumi/docs/blob/26a033fb30a38ebc19eeac0b395ad381c597b713/assets/sass/styles.scss#L30

File: content/docs/search-data.md [hunk 4]
```diff
@@ -1,7 +0,0 @@
----
-layout: "search-data"
```
Comment (justinvp) [inline content/docs/search-data.md:2]: The associated `layouts/docs/search-data.json` file should also be deleted.

File: content/docs/search-data.md [hunk 5]
```diff
@@ -1,7 +0,0 @@
...
-outputs: ["json"]
-exclude_from_pulumi_search_index: true
```
Comment (justinvp) [inline content/docs/search-data.md:4]: We should do a scrub of all files and remove use of `exclude_from_pulumi_search_index`.

File: scripts/build-search-index.js [hunk 6]
```diff
@@ -1,40 +0,0 @@
-// Adapted from https://lunrjs.com/guides/index_prebuilding.html
-
-const lunr = require("../static/js/lunr.min.js");
```
Comment (justinvp) [inline scripts/build-search-index.js:3]: `static/js/lunr.min.js` can be deleted.

**Review by justinvp — COMMENTED** [review]
🎉 Left some notes on further cleanup.

File: config/_default/config.toml [hunk 7]
```diff
@@ -6,6 +6,9 @@ sectionPagesMenu = "main"
 pygmentsCodeFences = true
 pygmentsUseClasses = true
 
+ignoreFiles = [ "content/docs/reference/pkg/*" ]
+refLinksErrorLevel = "WARNING"
```
Comment (cnunciato) [inline config/_default/config.toml:10]: Oops. 😇  I'll remove these.

**Comment by davidwrede** [comment]
Fixes #5250

**Comment by davidwrede** [comment]
@cnunciato Looks like the yarn.lock merge conflict is related to https://github.com/pulumi/docs/pull/4673 getting merged in. Do we need to take your updates, or should we resolve by merging in the changes from master?

**Comment by cnunciato** [comment]
@davidwrede Ack, those package.json additions were checked in by mistake. I'll back them out. Sorry about that.

**Comment by cnunciato** [comment]
All right, that should be done. Again, apologies for messing you up there.

**Comment by davidwrede** [comment]
@cnunciato No worries. Thanks!

@susanev RE https://github.com/pulumi/docs/pull/5223#issuecomment-778375604, with the latest change to moving the search box over to the top of the left nav, I made the input box pinned/sticky while scrolling the left nav. 

https://user-images.githubusercontent.com/1634752/108291290-e515e980-7146-11eb-93e0-21c559fcac2e.mp4

This feels like a better experience than what you showed above. Thoughts?

**Comment by susanev** [comment]
this is good, not sure what ur trying to call out as being different though.

[nit] could we add just a bit of pixels between the input and the scroll bar, so reduce the width just a bit?

**Comment by davidwrede** [comment]
The subtle difference is that the auto-complete scroll results don't just scroll to the top of the screen (and out of the doc page) as in what you observed earlier.

And yeah, I'm sure we can reduce the width a bit

**Review by susanev — APPROVED** [review]

**Comment by cnunciato** [comment]
@susanev @davidwrede Ok here's how this looks now -- the search box and results now scroll with the rest of the content in the left-hand side. Let me know what you think.

https://user-images.githubusercontent.com/274700/108433068-392dd600-71fa-11eb-8b7d-75b7512406e2.mov

**Comment by susanev** [comment]
@cnunciato looks good to me!

**Comment by davidwrede** [comment]
@cnunciato Yup, I like the new changes as well. :)

**Pull request merged**